ipatests/test_integration/test_commands.py

```diff
@@ -1814,6 +1814,37 @@ def test_sequence_processing_nonposix_group(self, cleanupgroups):
                                           '/tmp/reproducer2_code.py'])
         assert "missing attribute" not in result.stdout_text
 
+    def test_sidgen_task_continue_on_error(self):
+        """Verify that SIDgen task continue even if it fails to assign sid
+
+        related: https://pagure.io/freeipa/issue/9618
+        """
```

Comment:
In order to fully test the fix, the scenario has to be a bit more elaborate:
- add a user with no uid (it will be auto-assigned inside the range)
- add a user with uid 2000
- add a user with no uid (it will be auto-assigned inside the range)
- edit the first and 3rd users, remove the objectclass ipaNTUserAttrs and the attribute ipaNTSecurityIdentifier
- run the sidgen task
- ensure that user1 and user3 have a ipaNTSecurityIdentifier

Otherwise the test checks that there is a failure but not that the task continues with the other users.

As a side note, config-mod --add-sids should wait for the task to complete, I don't think the test needs to sleep 10s (https://pagure.io/freeipa/blob/024d41ebeaa875d500050aad39220d68eb70a709/f/ipaserver/install/adtrustinstance.py#_530-545)

Reply:
I assumed wrongly the task name wouldn't be predictable. It's that way most everywhere else in IPA.